InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.Modify.cs

```diff
@@ -0,0 +1,66 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Force.DeepCloner;
+using InternTrack.Core.Api.Models.Interns;
+using Moq;
+using Xunit;
+
+namespace InternTrack.Core.Api.Tests.Unit.Services.Foundations.Interns
+{
+    public partial class InternServiceTests
+    {
+        [Fact]
+        public async Task ShouldModifyInternAsync()
+        {
+            // given
+            int randomNumber = GetRandomNumber();
+            int randomDays = randomNumber;
+            DateTimeOffset randomDate = GetRandomDateTime();
+            Intern randomIntern = CreateRandomIntern();
+            Intern inputIntern = randomIntern;
+            Intern afterUpdateStorageIntern = inputIntern;
+            Intern expectedIntern = afterUpdateStorageIntern;
+            Intern beforeUpdateStorageIntern = randomIntern.DeepClone();
+            inputIntern.UpdatedDate = randomDate;
+            Guid internId = inputIntern.Id;
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectInternByIdAsync(internId))
+                    .ReturnsAsync(beforeUpdateStorageIntern);
```

Comment:
we need to get rid of `beforeUpdateStorageIntern` and simply return `updatedIntern`. While deepclone `expectedIntern`.